### PR TITLE
Add all relevant testcases for Arm Ethos-U85

### PR DESCRIPTION
**Pull request by zingo**
Add separate tests for Ethos-U85 to all backend operator tests.
Updated ethos-u-vela version to support more operators.

Signed-off-by: Per Åstrand <[per.astrand@arm.com](mailto:per.astrand@arm.com)>
Signed-off-by: Tom Allsop <[tom.allsop@arm.com](mailto:tom.allsop@arm.com)>

### CODE REVIEW

File: backends/arm/test/models/test_mobilenet_v2_arm.py
Comment (digantdesai) [inline backends/arm/test/models/test_mobilenet_v2_arm.py:101]: share with u55? And run it on corstone in the future?

File: examples/arm/setup.sh
Comment (digantdesai) [inline examples/arm/setup.sh:219]: In the summary can we add if there is any major change with this rev? Like one or two line would be helpful to understand the rationale.

File: backends/arm/test/ops/test_add.py
Comment (digantdesai) [inline backends/arm/test/ops/test_add.py:139]: how would this run u85 bitstream?

File: backends/arm/test/ops/test_linear.py
Comment (digantdesai) [inline backends/arm/test/ops/test_linear.py:176]: I am not sure if I am following the pattern where we enable/disable these. Is this operator specific? I would rather tie it to the ip i.e. something like this, hypothetically

```
if common.is_option_enabled("corstone300") and compile_spec.ip == "u55":
```

**Review by digantdesai — COMMENTED** [review]
Let's fix some small issues?

File: backends/arm/test/ops/test_add.py
Comment (per) [reply backends/arm/test/ops/test_add.py:139]: The compilespec is for Ethos-U55 and the run_method_and_compare_outpus is only run here for.

File: backends/arm/test/ops/test_avg_pool.py
Comment (per) [reply backends/arm/test/ops/test_avg_pool.py:155]: The comment on line 134/138 (# Expected to fail since ArmQuantizer cannot quantize a AvgPool2D layer) covers both tests.

File: backends/arm/test/ops/test_conv.py
Comment (per) [reply backends/arm/test/ops/test_conv.py:333]: We only serialize if the test should be running on FVP. Since it's time consuming to run them all (and on an additional coming target) we probably want to run on FVP for full models only, but not decided internally yet.

File: backends/arm/test/ops/test_depthwise_conv.py
Comment (per) [reply backends/arm/test/ops/test_depthwise_conv.py:226]: This is wrong and should be fixed. Sorry about that.

File: backends/arm/test/ops/test_expand.py
Comment (per) [reply backends/arm/test/ops/test_expand.py:109]: No, it's an internal bug that is in progress to be sorted out in Vela.

File: backends/arm/test/ops/test_linear.py
Comment (per) [reply backends/arm/test/ops/test_linear.py:176]: the `common.is_option_enabled("corstone300")` only says if the FVP is installed on the target and can be run. 
Right now it's only added for a few operators, but due to the time it adds to CI we probably want to move away from enabling it for all operator unit tests all the time.

File: examples/arm/setup.sh
Comment (per) [reply examples/arm/setup.sh:219]: Yes, good idea!

File: backends/arm/test/ops/test_conv.py
Comment (per) [reply backends/arm/test/ops/test_conv.py:318]: Good point. Will fix.

File: backends/arm/test/models/test_mobilenet_v2_arm.py
Comment (zingo) [reply backends/arm/test/models/test_mobilenet_v2_arm.py:101]: > share with u55? 

How do you mean? We already have a test_mv2_u55_BI() ?

> And run it on corstone in the future?

Maybe we are trying out different ways to run larger models like mv2 and others in a different way then the unit test setup. Lest see how it turns out :)

File: backends/arm/test/ops/test_depthwise_conv.py
Comment (zingo) [reply backends/arm/test/ops/test_depthwise_conv.py]: pul request updated

File: backends/arm/test/ops/test_conv.py
Comment (zingo) [reply backends/arm/test/ops/test_conv.py]: pull request updated

File: backends/arm/test/ops/test_depthwise_conv.py
Comment (zingo) [inline backends/arm/test/ops/test_depthwise_conv.py:134]: This tests is causing problem in your CI dw_conv2d_3x3_1x3x256x256_gp3_st1 but works when we test it, lets remove it again (it was added back last week) for now and fix that in a separate patch after all is in.

Comment (zingo) [reply backends/arm/test/ops/test_depthwise_conv.py:134]: As a note the test running this was before this patch marked as @unittest.expectedFailure  but that line was removed below in this patch.

File: backends/arm/test/models/test_mobilenet_v2_arm.py
Comment (digantdesai) [reply backends/arm/test/models/test_mobilenet_v2_arm.py:101]: Share the body of the test I mean, not a big deal TBH

File: backends/arm/test/ops/test_add.py
Comment (digantdesai) [reply backends/arm/test/ops/test_add.py:139]: Right, but this function can be called for both u55 and u85 IIUC

File: backends/arm/test/ops/test_linear.py
Comment (digantdesai) [reply backends/arm/test/ops/test_linear.py:176]: IIUC this FVP is u55 specific while this function can also get u85 compile spec and vela/et can produce u85 binary. I was trying to highlight that the choice of fvp 300 vs 320 should be tied to the compile spec somehow.

**Review by digantdesai — APPROVED** [review]
Stamping to unblock.

**Comment by zingo** [comment]
Just a try to rebase it on latest after Fredriks pull request got merged if you did this already on the "imported" please ignore the update.

**Pull request closed without merging**
